src/schemas/task.json

```diff
@@ -17,21 +17,26 @@
       "type": "string"
     },
     "OutputPath": {
-      "type": ["string", "null"]
+      "$ref": "paths.json#/definitions/asl_path"
     },
     "InputPath": {
-      "type": ["string", "null"]
+      "$ref": "paths.json#/definitions/asl_path"
     },
     "Resource": {
-      "oneOf": [{
-        "type": "string",
-        "pattern": "^(arn:aws:([a-z]|-)+:([a-z]|[0-9]|-)*:[0-9]*:([a-z]|-)+:[a-zA-Z0-9-_.]+(:(\\$LATEST|[a-zA-Z0-9-_]+))?$)|(\\${.*\\})"
-      }, {
-        "type": "object"
-      }]
+      "type": "string",
+      "oneOf": [
+        {
+          "$comment": "defers validation to the registered formatter",
+          "format": "asl_arn"
+        },
+        {
+          "$comment": "allows for variable substitution",
+          "pattern": "^\\$\\{[^\\}]+\\}$"
+        }
+      ]
```

Comment:
easier to manage the more complex regex in code rather than the schema's limited regex